sonarts-core/src/utils/sonarAnalysis.ts

```diff
@@ -132,10 +132,10 @@ export class SonarIssue extends tslint.RuleFailure {
     return this;
   }
 
-  public addSecondaryLocation(node: ts.Node, message: string): SonarIssue;
+  public addSecondaryLocation(node: ts.Node, message?: string): SonarIssue;
   public addSecondaryLocation(secondaryLocation: IssueLocation): SonarIssue;
   public addSecondaryLocation(): SonarIssue {
```

Comment:
I haven't realized I can use `instanceof` this way. Then we can make this more explicit with signature like 
`addSecondaryLocation(nodeOrLocation: ts.Node | IssueLocation, message?: string): SonarIssue `  and avoid using `arguments`